shoulder/model/field.py

```diff
@@ -27,11 +27,41 @@
 class Field():
     """ Models a single named field (or bitfield) in a register fieldset """
 
-    name: str
-    """ Name of the field """
+    name: str = ""
+    """ The abbreviated/symbolic/accronym name for this field """
 
-    msb: int
+    msb: int = 0
     """ Most significant bit that the field occupies within a register """
 
-    lsb: int
+    lsb: int = 0
     """ Least significant bit that the field occupies within a register """
+
+    long_name: str = ""
+    """ The non-abbreviated/spelled out name for this field """
+
+    description: str = ""
+    """ A description of this field """
+
+    readable: bool = False
+    """ True if this field is readable  """
+
+    writable: bool = False
+    """ True if this field is writable  """
+
+    lockable: bool = False
+    """ True if this field can be locked  """
+
+    write_once: bool = False
+    """ True if this field becomes locked after writing to it once  """
+
+    write_1_clear: bool = False
+    """ True if this field is cleared (set to '0') by writing a '1' to it  """
+
+    reserved0: bool = False
+    """ True if the value of this field is reserved 0  """
+
+    reserved1: bool = False
+    """ True if the value of this field is reserved 1  """
+
+    preserved: bool = False
+    """ True if the value of this field should be preserved across writes  """
```

Comment:
This is a really great point to bring up, and I'm not sure I have a simple answer for this. On one hand, you're absolutely right that the attributes above aren't explicitly defined in the ARM manuals. On the other hand, these attributes are describing an architecture-agnostic *usage* of a field, and (in my mind) the *usage* of a field doesn't really belong to an architecture.

I'm inclined to leave these attributes here, but could be convinced otherwise if you have any more reasons for moving them.